### DISCUSSION TITLE
phase field dissolution

### DISCUSSION THREAD

**Discussion by latmarat** [idaholab/moose#21637] [Q&A Modules: Phase field] [answered]
Hello: 

I am interested in simulating dissolution of pre-populated particles (2D or 3D) in a matrix using MOOSE/phase-field. Ideally, I would like to use thermodynamic functions from CALPHAD models. Do you have some example problems/input files that could be good starting points for such simulations in MOOSE? 

Thanks!

**Comment by GiudGiud** [comment]
@SudiptaBiswas @laagesen

**Comment by laagesen** [comment]
Hello- are you wanting to do this for an alloy system? Like say an ordered intermetallic phase that is dissolving into a random solid solution phase in response to a change in temperature?

**Reply by latmarat** [reply]
@laagesen yes! that's exactly what I would like to do :)

**Comment by laagesen** [comment, marked as answer]
OK got it. We don't have an example that does exactly this in the repo, so I think the easiest thing to do would be to adapt one of the existing examples that simulates two-phase coexistence and modify it. We have a couple examples where you have an order parameter representing the phase and a single composition variable. In these cases, there is two-phase coexistence as dictated by the common tangent construction between the two phase free energies. I guess the simplest thing to do would be to add a constant offset to the intermetallic phase free energy so that it is always above the random solid solution phase free energy, so that dissolution of the intermetallic phase would be energetically favorable. I'd suggest trying this out with simple parabolic free energies before moving to Calphad-based functions; try doing two-phase coexistence, make sure you get the equilibrium compositions you expect first then add an offset to force dissolution.

I think the best starting places would be: for a WBM-type phase-field model, there is a test you could modify:
moose/modules/phase_field/test/tests/MultiPhase/derivativetwophasematerial.i
and you can see the documentation for this type of model here:
https://mooseframework.inl.gov/source/materials/DerivativeTwoPhaseMaterial.html#!
There are some tweaks to this test if you want to use it for a physical system, such as increasing mesh resolution; let's discuss further if you want to go this route.

For a KKS-type of phase field model, you could start from
moose/modules/phase_field/examples/kim-kim-suzuki/kks_example_noflux.i
This is a 1D example but the parameters are maybe a little closer to simulating a physical system. You can find documentation on the KKS models here:
https://mooseframework.inl.gov/modules/phase_field/MultiPhase/KKS.html
